### ISSUE TITLE
Some doubt of reset()

### ISSUE THREAD

**Issue by luan-g** [NVIDIA/DALI#1266]
Hi, thanks for your contrubution of building this package. Could you please help me with the following problems?
I want to use `DALIClassificationIterator` to load data in order. And I close `random_shuffle` and `shuffle_after_epoch`  in `ops.FileReader`.
In first epoch, the data is ordered. But when I use `reset()` method of  `DALIClassificationIterator` after first epoch, the order is shuffled. 
And I find if I set `bathsize=1` the data is ordered even after `reset()` .
Is there any bug for `reset()`?

Now I can only load ordered data by redoing  `dataloader=DALIClassificationIterator(...)` after every epoch. Is there any other ways to realize this?

**Comment by awolant** [comment]
Hi, thanks for the question.
Could you provide some code sample of your pipeline and data loader?
Is your dataset size divisible by batch size you are using? If no, what values of `fill_last_batch` and 'last_batch_padded' are you using in the data loader?

**Comment by luan-g** [comment]
Your code have some bug, because the `b[0]` is changeable. I have fixed it in below code.

There is the complete code. I changed the code to close randomcrop and flip.
```python
import time
import numpy as np
import nvidia.dali.ops as ops
import nvidia.dali.types as types
from nvidia.dali.pipeline import Pipeline
from nvidia.dali.plugin.pytorch import DALIClassificationIterator, DALIGenericIterator

class HybridTrainPipe(Pipeline):
    def __init__(self, batch_size, num_threads, device_id, data_dir, crop, dali_cpu=False, local_rank=0, world_size=1, shuffle=False, file_list=''):
        super(HybridTrainPipe, self).__init__(batch_size, num_threads, device_id, seed=12 + device_id)
        dali_device = "gpu"
        self.input = ops.FileReader(file_root=data_dir, shard_id=local_rank, num_shards=world_size, shuffle_after_epoch=shuffle,file_list=file_list)
        self.decode = ops.ImageDecoder(device="mixed", output_type=types.RGB)
        #self.res = ops.RandomResizedCrop(device="gpu", size=crop, random_area=[0.08, 1.])
        self.res = ops.Resize(device="gpu", resize_x=crop, resize_y=crop)
        self.cmnp = ops.CropMirrorNormalize(device="gpu",
                                            output_dtype=types.FLOAT,
                                            output_layout=types.NCHW,
                                            image_type=types.RGB,
                                            mean=[0.485 * 255, 0.456 * 255, 0.406 * 255],
                                            std=[0.229 * 255, 0.224 * 255, 0.225 * 255])
        self.coin = ops.CoinFlip(probability=0.5)

    def define_graph(self):
        #rng = self.coin()
        self.jpegs, self.labels = self.input(name="Reader")
        images = self.decode(self.jpegs)
        images = self.res(images)
        output = self.cmnp(images)
        return [output, self.labels]
```


```python
train_loader = get_imagenet_iter_dali(type='train', image_dir=data_dir, batch_size=5,
                                          num_threads=8, crop=224, device_id=0, shuffle=False)
import copy
d1 = []
for b in train_loader:
    # d1.append(b[0])
    d1.append(copy.deepcopy(b[0]))

train_loader.reset()

d2 = []
for b in train_loader:
    # d2.append(b[0])
    d2.append(copy.deepcopy(b[0]))

for b1, b2 in zip(d1, d2):
    assert (b1['data'] == b2['data']).all()
```

I chosen 256 images to debug.
In the above code, when dataset size is divisible by batch size for example `batch_size=4`, the `assert` can pass, while  dataset size is not divisible by batch size for example `batch_size=5`, the `assert` is error.

I have some further found.
```python
train_loader = get_imagenet_iter_dali(type='train', file_list='idx.txt', image_dir=data_dir, batch_size=5,
                                          num_threads=8, crop=224, device_id=0, shuffle=False)
```
I set the order number as class in the `file_list='idx.txt'` ,such as the first image is 0, the second image is 1, the last is 255.  I found the images is ordered, but it doesn't begin with 0. For example, `d2[0]['label']` is 4,5,6,7,8, while `d1[0]['label']` is 0,1,2,3,4

Maybe, it is due to `reset()` doesn't reset the index in DALI  somewhere.

**Comment by JanuszL** [comment]
@luan-g - we have managed to repro this problem. We will check it and get back to you soon.

**Comment by JanuszL** [comment]
@luan-g - to get expected behaviour you need to set `last_batch_padded=true` in the `DALIClassificationIterator`, and `pad_last_batch=true` in the FileReader as well.
The first variable tells the iterator that the remainder of the last batch is a dummy load and after resetting the number of samples consumed from the next epoch should be set to 0. The second one tells the reader to provide the dummy data. They need to be set manually as you may have more than one reader in the pipeline and for some reason, you may want to have a different behavior of each of them.
On top of that, I have extended Framework iterators docs by that https://github.com/NVIDIA/DALI/pull/1351.

**Comment by luan-g** [comment]
I can get the expected behaviour. Thanks for your reply.
